op-node/rollup/engine/payload_success.go

```diff
@@ -51,5 +52,46 @@ func (e *EngineController) onPayloadSuccess(ctx context.Context, ev PayloadSucce
 	err := e.tryUpdateEngineInternal(ctx)
 	if err != nil {
 		e.log.Error("Failed to update engine", "error", err)
+	} else {
+		updateEngineFinish := time.Now()
+		e.logBlockProcessingMetrics(updateEngineFinish, ev)
 	}
 }
+
+func (e *EngineController) logBlockProcessingMetrics(updateEngineFinish time.Time, ev PayloadSuccessEvent) {
+	// Protect against nil pointer dereferences
+	if ev.Envelope == nil || ev.Envelope.ExecutionPayload == nil {
+		e.log.Debug("Envelope.ExecutionPayload not found, skipping block processing metrics")
+		return
+	}
+
+	buildTime := time.Duration(0)
+	insertTime := updateEngineFinish.Sub(ev.InsertStarted)
+	totalTime := insertTime
+
+	// BuildStarted may be zero if sequencer already built + gossiped a block, but failed during
+	// insertion and needed a retry of the insertion. In that case we use the default values above,
+	// otherwise we calculate buildTime and totalTime below
+	if !ev.BuildStarted.IsZero() {
+		buildTime = ev.InsertStarted.Sub(ev.BuildStarted)
+		totalTime = updateEngineFinish.Sub(ev.BuildStarted)
+	}
+
+	// Protect against divide-by-zero
+	var mgasps float64
+	if totalTime > 0 {
+		// Calculate "block-processing" mgasps.
+		// NOTE: "realtime" mgasps (chain throughput) is a different calculation: (GasUsed / blockPeriod)
+		mgasps = float64(ev.Envelope.ExecutionPayload.GasUsed) * 1000 / float64(totalTime)
```

Comment:
nit: might it be better like so?

```suggestion
		// MGas/s; avoids relying on Duration’s nanosecond representation
mgasps := float64(ev.Envelope.ExecutionPayload.GasUsed) / 1e6 / totalTime.Seconds()
```

* Reads a bit easier (at least, to me)
* Fewer conversions (so more performant)[totalTime.Seconds() already returns float64, so you avoid converting the Duration yourself (one less float64(...))]
* May improve precision as it avoids floating-point multiply by 1000 on a large mantissa (not sure if this is actually an issue here, mind you)

Reply:
Yes good call. Will change to your suggestions